pkg/network/tracer/testutil/proxy/unix_transparent_proxy.go

```diff
@@ -154,16 +154,22 @@ func (p *UnixTransparentProxyServer) handleConnection(unixSocketConn net.Conn) {
 			unixReader := bufio.NewReader(unixSocketConn)
 			remoteReader := bufio.NewReader(remoteConn)
 
+			// Reuse this buffer instead of allocating new ones every time both
+			// for efficiency and to attempt to work around some issues with
+			// page migration kicking in and leading to minor page faults in TLS
+			// probes and flaky tests on some kernel versions.
+			big := make([]byte, 1024*1024)
+
 			for {
-				buf := make([]byte, 8)
+				buf := big[0:8]
 				_, err := io.ReadFull(unixReader, buf)
 				if err != nil {
 					break
 				}
 				readSize := binary.BigEndian.Uint64(buf)
 
 				if readSize != 0 {
```

Comment:
edge case - verify readSize is not bigger than 1MB (the size of `big`)

Reply:
same for line 194

Reply:
Done.